WordPress/gutenberg
# Patterns: add a hint about the rename of reusable blocks to menu and inserter

## What?
Adds a hint to the menu and tab inserter to alert users to rename of reusable blocks

## Why?
To help users with the transition

## How?
Uses the existing preferences infrastructure to allow hint to be dismissed

## Testing Instructions

- Add a new block
- In the block overflow menu make sure  the menu is called `Create pattern/reusable block` 
- In the modal make sure a hint about reusable block renaming appears, don't dismiss it
- In the Inserter `Synced patterns` tab check that the hint also shows
- Dismiss the hint and make sure it doesn't display in either location even after browser reload

## Keyboard Testing Instructions

1. Add a block to the post editor
2. Shift tab to the block toolbar
3. Use the right arrow key until you get to the options menu and open it.
4. Arrow down through the options, and note that there's now a 'hint' within the menu that can be dismissed.

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/3629020/7973f23c-ad13-4c6b-8b83-851b089f6c5e
